lib/routes/gdufs/xwxy/index.ts

```diff
@@ -0,0 +1,250 @@
+import { parseDate } from '@/utils/parse-date';
+
+const handler = async (ctx) => {
+    const BASE_URL = 'https://xwxy.gdufs.edu.cn';
+
+    const category = ctx.req.param('category') ?? 'news';
+    const pathMap: Record<string, string> = {
+        news: '/xwzx/xyxw.htm',
+        notices: '/xwzx/tzgg/tz.htm',
+        announcements: '/xwzx/tzgg/gg.htm',
+        media: '/xwzx/mtjj.htm',
+    };
+    const titleMap: Record<string, string> = {
+        news: '学院新闻',
+        notices: '通知',
+        announcements: '公告',
+        media: '媒体聚焦',
+    };
+
+    const link = `${BASE_URL}${pathMap[category] ?? pathMap.news}`;
+
+    const response = await got(link);
+    if (!response.body) {
+        throw new Error('No response body');
+    }
+    const $ = load(response.body);
+    // 列表选择器：优先正文区域
+    let list = $('section.TextList ul li a[href]');
+    if (list.length === 0) {
+        list = $('div.flex-center a.clearfix');
+    }
+    if (list.length === 0) {
+        list = $('div.flex-center a[href]');
+    }
+    if (list.length === 0) {
+        list = $('ul li a[href]');
+    }
+    if (list.length === 0) {
+        list = $('a[href]');
+    }
+
+    // 仅保留指向详情页、外部媒体或内部附件的链接
+    const anchors = list
+        .toArray()
+        .map((el) => $(el))
+        .filter((a) => {
+            const href = (a.attr('href') || '').trim();
+            if (!href || href === '#' || href.startsWith('javascript:') || href.startsWith('mailto:') || href.startsWith('data:') || href.startsWith('vbscript:')) {
+                return false;
+            }
+            // 明确放行详情与外链
+            if (href.includes('/info/') || href.startsWith('http')) {
+                return true;
+            }
+            // 放行常见内部附件与资源路径
+            const assetLike = href.startsWith('_upload/') || href.startsWith('/_upload/') || /\.(pdf|docx?|xlsx?|pptx?|zip|rar|jpg|jpeg|png|gif|mp4|htm|html)([#?].*)?$/i.test(href);
+            if (assetLike) {
+                return true;
+            }
+            // 回退：解析为绝对 URL 后按主机名判断为本站
+            try {
+                const u = new URL(href, link);
+                return u.hostname.endsWith('gdufs.edu.cn');
+            } catch {
+                return false;
+            }
+        });
+
+    const items = anchors
+        .map((a) => {
+            const href = a.attr('href') || '';
+            const li = a.closest('li');
+            const contextText = ((li && li.text()) || a.text()).trim();
+            const datePattern = /\d{4}[-/.]\d{1,2}[-/.]\d{1,2}/;
+            const dateText = a.find('i').text().trim() || (li && li.find('i').text().trim()) || (li && li.find('time').text().trim()) || (contextText.match(datePattern)?.[0] ?? '');
+            const pubDate: Date | undefined = dateText ? parseDate(dateText) : undefined;
+            const title = a.find('p').text().trim() || a.find('h5').text().trim() || a.attr('title')?.trim() || a.text().trim() || contextText.replace(datePattern, '').trim();
```

Comment:
Do not use `p` as it does not select the title in https://github.com/DIYgod/RSSHub/pull/20092#issuecomment-3305114215